SanitizerTest not fully implemented #5048

YongJieYongJie opened the pull request:
Fixes #5048

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -87,16 +87,16 @@ public static String sanitizeTextField(String rawText) {
+     * @param string
+     * @return the sanitized string or null (if the parameter was null).
      */
     public static String sanitizeForJs(String str){ 
```
Since you're already changing this, space before ```{```

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -87,16 +87,16 @@ public static String sanitizeTextField(String rawText) {
+     * @return the sanitized string or null (if the parameter was null).
      */
     public static String sanitizeForJs(String str){ 
+        if(str == null) return null;
```
Please refer to our Java coding style guide at https://docs.google.com/document/pub?id=1iAESIXM0zSxEa5OY7dFURam_SgLiSMhPQtU0drQagrs&embedded=true

For your ease -

![screen shot 2016-03-22 at 8 41 32 pm](https://cloud.githubusercontent.com/assets/6291947/13975026/869eb656-f06e-11e5-8c82-98a84654ed4b.png)

Please fix

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -120,6 +120,10 @@ public static String sanitizeForHtml(String str){
                 .replaceAll("&(?!(amp;)|(lt;)|(gt;)|(quot;)|(#x2f;)|(#39;))", "&amp;");
     }
     
+    public static String sanitizeHtmlForSaving(String html) {
```
This method is pretty odd - it just calls sanitizeForHtml, is this being used often?

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -142,8 +146,12 @@ public static String sanitizeForUri(String uri) {
+     * @return the sanitized url or null (if the parameter was null).
      */
     public static String sanitizeForNextUrl(String url) {
+        if(url == null) return null;
```
Same here - coding style

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -186,8 +194,8 @@ public static String sanitizeForSearch(String str){
      */
-    public static String sanitizeForCsv(String str) {
-        return "\"" + str.replace("\"", "\"\"") + "\"";
+    public static String sanitizeForCsv(String str){ 
```
Revert the space

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -186,8 +194,8 @@ public static String sanitizeForSearch(String str){
-    public static String sanitizeForCsv(String str) {
-        return "\"" + str.replace("\"", "\"\"") + "\"";
+    public static String sanitizeForCsv(String str){ 
+        return "\"" + str.replace("\"", "\"\"") + "\""; // in CSV, " is escaped as ""
```
I don't think the comment is necessary, if someone were to come and modify this method they'd probably research more about sanitizing for CSV and know that

If you think it's an obscure thing and should be commented, add it into the method comment and add it as a "note" rather than here

junhaoyap commented on src/test/java/teammates/test/cases/common/SanitizerTest.java:
```diff
@@ -44,27 +47,62 @@ public void testSanitizeTextField() {
     @Test
     public void testSanitizeForJs() {
-        
+
```
Unnecessary blank lines after a method - let's remove all these to keep it slightly shorter

junhaoyap commented on src/test/java/teammates/test/cases/common/SanitizerTest.java:
```diff
@@ -44,27 +47,62 @@ public void testSanitizeTextField() {
     @Test
     public void testSanitizeForRichText() {
-        
+        // using org.apache.commons.lang3.StringEscapeUtils.escapeHtml4()
```
Can we rephrase this slightly to ```Not tested - using org.apache.commons.lang3.StringEscapeUtils.escapeHtml4()```

junhaoyap commented:
@YongJieYongJie mainly style comments - I do have one more major comment and let's fix those before we do another round of review

junhaoyap commented on src/test/java/teammates/test/cases/common/SanitizerTest.java:
```diff
@@ -44,27 +47,62 @@ public void testSanitizeTextField() {
+        String sanitized = Sanitizer.sanitizeForHtml(unsanitized);
+        String sanitizedTwice = Sanitizer.sanitizeForHtml(sanitized);
+
+        assertEquals(null, Sanitizer.sanitizeForHtml(null));
```
This comment applies to all the different tests that you've written - can we break up the different tests within a parent test into "lesser test methods" so it is more modularized and it's immediately apparent what each "lesser test method" is trying to test for 

For instance, ```assertEquals(null, Sanitizer.sanitizeForHtml(null));``` would go under something like ```testSanitizeForHtmlReceivesNullReturnsNull```

and so on, single sanitization, double sanitization etc.

junhaoyap replied:
https://docs.google.com/document/pub?id=1iAESIXM0zSxEa5OY7dFURam_SgLiSMhPQtU0drQagrs&embedded=true

Also you can optionally adhere to this naming method if it will help to make things more clear as I understand the current naming method might not be as nice sounding

![screen shot 2016-03-22 at 9 01 07 pm](https://cloud.githubusercontent.com/assets/6291947/13975348/514cc0c6-f071-11e5-9db0-c2bbd84fa108.png)

junhaoyap commented:
@YongJieYongJie Please fix the relevant stuff and do get back to me! Looking forward :+1:

YongJieYongJie commented:
@junhaoyap I've fixed the above mentioned issues.

Specifically, I
- applied Java coding style to modified code
- applied Java coding style to existing code (which were quite stale and inconsistent with Java coding style)
- split tests into smaller modules
- removes redundant `sanitizeHtmlForSaving()`, which were only used in one place

junhaoyap commented on src/main/java/teammates/common/util/Sanitizer.java:
```diff
@@ -108,8 +114,10 @@ public static String sanitizeForJs(String str){
-    public static String sanitizeForHtml(String str){ 
-        if(str == null) return null;
+    public static String sanitizeForHtml(String str) {
+        if(str == null) {
```
Space before ```(```

junhaoyap commented:
Much better - unfortunately it's very late here in California and I have to head to bed soon so I have to review this after work tomorrow, fix the last cosmetic issue I have and I will review it tomorrow, good job @YongJieYongJie

junhaoyap commented:
@YongJieYongJie looks good, @damithc please take a look

junhaoyap commented:
@YongJieYongJie merged - :100:
